File: ios/Packages/FeatureInventory/Sources/FeatureInventory/Module/InventoryModule.swift
import Contracts
import WireCore

/// Inventory feature module — accessible to ADMIN and STAFF.
public final class InventoryModule: AppModule {
    public init() {}

    public let metadata = ModuleMetadata(
        id: "inventory",
        name: "Inventory",
        requiredRoles: [.ADMIN, .STAFF],
        priority: 85
    )

    public func initialize(context: any ModuleContext) {
        for slot in provideWidgets() {
            context.slotRegistry.register(slot)
        }
    }

    public func provideRoutes() -> [ModuleRoute] {
        [ModuleRoute(route: "/inventory", requiredRole: .STAFF)]
    }
}

File: flutter/feature_inventory/lib/src/module/inventory_module.dart
import 'package:contracts/contracts.dart';

/// Inventory feature module — accessible to ADMIN and STAFF.
class InventoryModule extends AppModule {
  @override
  ModuleMetadata get metadata => const ModuleMetadata(
        id: 'inventory',
        name: 'Inventory',
        requiredRoles: {Role.ADMIN, Role.STAFF},
        priority: 85,
      );

  @override
  void initialize(ModuleContext context) {
    for (final slot in provideWidgets()) {
      context.slotRegistry.register(slot);
    }
  }

  @override
  List<ModuleRoute> provideRoutes() => [
        const ModuleRoute(route: '/inventory', requiredRole: Role.STAFF),
      ];
}

